src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -466,6 +471,38 @@ static int open_device(const char* path, bool try_load_moudle = false)
   return nbd;
 }
 
+static bool find_key_from_conf_file(const char *key)
+{
+  std::vector <std::string> my_sections;
+  g_conf->get_my_sections(my_sections);
+  std::string admin_socket_str;
+  return ((g_conf->get_val_from_conf_file(my_sections, key,
+           admin_socket_str, false) == 0) ? true : false);
```

Comment:
Ah, "/sys/block/nbdX/pid" really makes the life much easier! I thought there should be something like this and looked for it some time ago, but failed to find then.

I agree that there is no need in asok for `list-mapped` in this case, and now I don't have a strong opinion about the default for admin-socket (both empty and  {run_dir}/rbd-nbd.{nbd_dev}.asok look good to me).